homeassistant/components/evohome/__init__.py

```diff
@@ -215,10 +213,12 @@ def init_client(self) -> bool:
         store = self.hass.helpers.storage.Store(STORAGE_VERSION, STORAGE_KEY)
         await store.async_save(self._app_storage)
 
+        access_token_expires_utc += self.params[CONF_SCAN_INTERVAL]
```

Comment:
why do we add scan interval? That seems unrelated?

Reply:
It was to handle propagation/rounding errors in `access_token_expires`.

Adding this ensures that `self._save_auth_tokens()` is invoked via `async_track_point_in_utc_time()` only _after_ the token has actually expired.

It is no big problem if called `self._save_auth_tokens()` is invoked _before_ the token has actually expired, but it means that `async_track_point_in_utc_time()` is called multiple times, with the result of unnecessary I/O...

When I used a `scan_interval` of 60 sec and _with_ adding `self.params[CONF_SCAN_INTERVAL]`, I got 23 invocations of self._save_auth_tokens()` (1 every seconds) before the access token actually expired.

The downside of this 'trick' is that a new access token will not be saved if HA is halted in the time between a new token, and it being saved.

Thus, I think I make it 60s, or even 30s if you prefer.